frame/elections-phragmen/src/lib.rs

```diff
@@ -654,8 +658,12 @@ decl_module! {
 impl<T: Config> Module<T> {
 	/// The deposit value of `count` votes.
 	fn deposit_of(count: usize) -> BalanceOf<T> {
-		T::VotingBondBase::get().saturating_add(
-			T::VotingBondFactor::get().saturating_mul((count as u32).into())
+		T::VotingBondStorageBase::get().saturating_add(
+			T::VotingBondStorageFactor::get().saturating_mul((count as u32).into())
+		).saturating_add(
+			T::VotingBondWeightFactor::get().saturating_mul(
+				(T::WeightInfo::election_phragmen(1, (count as u32).into(), 0) as u32).into()
```

Comment:
Here you should not convert weight (u64) to u32 and then `.into()` it to balance. This can easily loose accuracy. 

Instead, you should directly convert weight to balance with `saturated_into`, see `SaturatedConversion` trait for reference.